third_party/terraform/resources/resource_container_cluster.go.erb

```diff
@@ -784,6 +787,32 @@ func resourceContainerCluster() *schema.Resource {
 	}
 }
 
+// Setting 0 for guest accelerator count has the effect of removing all of the guest_accelerators. Since the API
+// returns an empty block we are squelching the diff if the API block is empty but the count is zero.
+//
+// This could also be modelled by setting `guest_accelerator = []` in the config. However since the
+// previous syntax requires that schema.SchemaConfigModeAttr is set on the field it is advisable that
+// we have a work around for removing guest accelerators.
+// See https://github.com/terraform-providers/terraform-provider-google/issues/3786
+func resourceNodeConfigGuestAcceleratorCustomizeDiff(diff *schema.ResourceDiff, meta interface{}) error {
+	old, new := diff.GetChange("node_config.0.guest_accelerator")
+	oList := old.([]interface{})
+	nList := new.([]interface{})
+
+	if len(nList) == len(oList) {
+		return nil
+	}
+
+	if len(oList) == 0 && len(nList) > 0 {
```

Comment:
What if I've defined the following?


```hcl
guest_accelerator {
  count = 0
  type  = "nvidia-tesla-p100"
}

guest_accelerator {
  count = 0
  type  = "nvidia-tesla-p4"
}
```

Or this:

```hcl
guest_accelerator {
  count = 1
  type  = "nvidia-tesla-p100"
}

guest_accelerator {
  count = 0
  type  = "nvidia-tesla-p4"
}
```

I'd be more comfortable with suppressing the case where `count = 0` and `type = ""` I think. That matches what PSO's Terraform `0.11` module has defined more closely, and treats the singular empty and falsey block as a special case instead of any first block with a `count` of `0`.

Reply:
Good call, it's not handling the multiple accelerators great right now.

In regards to only ignoring empty types, I think your example above shows a possible reason why you would want to keep the type in the config. But I'm also ok with starting with the more restrictive option and expanding it if a non-theoretical need arises.